pennylane/qnode.py

```diff
@@ -524,6 +525,12 @@ def construct(self, args, kwargs):
         with self.qtape:
             self.qfunc_output = self.func(*args, **kwargs)
 
+        # If there are any custom operations defined for the device, now is the time.

```

Comment:
@josh146 applying the replace function here results in the decompositions defined in a device to be propagated through.